src/diffusers/image_processor.py

```diff
@@ -882,3 +883,56 @@ def preprocess(
             depth = self.binarize(depth)
 
         return rgb, depth
+
+
+class IPAdapterMaskProcessor(VaeImageProcessor):
+    """
+    Image processor for IP Adapter image masks.
+
+    """
+    def __init__(self):
+        super().__init__(do_normalize=False, do_binarize=True, do_convert_grayscale=True)
+
+    def process(self, images: Union[List[PIL.Image.Image], PIL.Image.Image]) -> torch.FloatTensor:
+        """
+        Convert a PIL.Image.Image or a list of PIL.Image.Image to a torch.FloatTensor
+        """
+        if not isinstance(images, list):
+            images = [images]
+        images = self.preprocess(images)
+        return images
+
+    @staticmethod
+    def downsample(mask: torch.FloatTensor, batch_size: int, seq_length: int, value_embed_dim: int):
+        """
+        Downsample a mask to target seq_length
+        """
+        o_h = mask.shape[1]
+        o_w = mask.shape[2]
+        ratio = o_w / o_h
+        mask_h = int(torch.sqrt(torch.tensor(seq_length / ratio)))
+        mask_h = int(mask_h) + int((seq_length % int(mask_h)) != 0)
+        mask_w = seq_length // mask_h
+
+        mask_downsample = F.interpolate(mask.unsqueeze(0), size=(mask_h, mask_w), mode="bicubic").squeeze(0)
+
+        # Repeat mask until batch_size
```

Comment:
(nit): "until" indicates that there could be a `while` in the subsequent operations. A better comment could be:

"# Repeat `batch_size` times". WDYT?